packages/block-editor/src/components/block-pattern-setup/index.js

```diff
@@ -170,6 +244,14 @@ const BlockPatternSetup = ( {
 					activeSlide={ activeSlide }
 					patterns={ patterns }
 					onBlockPatternSelect={ onPatternSelectCallback }
+					onChangeSlide={ ( newSlide ) => {
+						setActiveSlide(
+							Math.min(
+								Math.max( newSlide, 0 ),
+								patterns.length - 1
+							)
+						);
+					} }
```

Comment:
This allows us to jump to any slide from within the carousel block. The min/max clamp ensures we keep the active slide within bounds.